core/src/main/java/org/axonframework/eventsourcing/eventstore/BatchingEventStorageEngine.java

```diff
@@ -42,7 +42,8 @@
 public abstract class BatchingEventStorageEngine extends AbstractEventStorageEngine {
 
     private static final int DEFAULT_BATCH_SIZE = 100;
-    private final int batchSize;
+
+    protected final int batchSize;
```

Comment:
I am not too fond of making field more open.
Could it be an alternative to extract the `fetchUntilEmpty` being `true` to a protected method? Subclasses could then override that method to indicate that a partial batch does not mean a final batch.

Reply:
Nice, liking that idea. Wasn't feeling to comfortable about this any how.
I'll change it around, thanks for the suggestion!